src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py

```diff
@@ -85,15 +82,6 @@ def __init__(self, session):
 
         await self.shutdown_task_manager()
 
-        # kill all the tracker sessions.
-        # Wait for the defers to all have triggered by using a DeferredList
-        for tracker_url in self._session_list.keys():
-            for session in self._session_list[tracker_url]:
-                self.session_stop_task_list.append(session.cleanup())
```

Comment:
Doesn't removing this mean that we don't shutdown the TaskManager of each individual session?

Reply:
Each HTTPSession cleans itself on cancel and when it is done:
https://github.com/Tribler/tribler/blob/8fcba201280effb89dff424b557bbfa240abf16f/src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py#L316

Reply:
I could be wrong, but looking at `connect_to_tracker` it seems that if there is a `CancelledError`, it doesn't call `clean_session`.
https://github.com/Tribler/tribler/blob/8fcba201280effb89dff424b557bbfa240abf16f/src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py#L136-L147

Reply:
You're right, I'll fix it.

Reply:
added `.clean_session` call to `CancelledError`